### PR TITLE
Fix line numbers of input file

### LINKED ISSUE #19: line number in error msg is not correct
Use the following test file:

    select 1;
    select 1;
    
    # following drop SQL should give error, because there is no t1 existing.
    
    drop table t1;

Error msg said error happens in **line 2**. But actually it's in **line 6**. If we have very a very long test. It is very annoying to **manually** find the line number of the specific error every time an error occurs.

    INFO[0000] recording tests: [example]
    WARN[0000] Create new db&{0 {root:root@tcp(127.0.0.1:3307)/test?strict=true&time_zone=%27Asia%2FShanghai%27&allowNativePasswords=true&allowAllFiles=true 0x1068278} 0 {0 0} [0xc0001a07e0] map[] 0 1 0xc0000781e0 false map[0xc0001a07e0:map[0xc0001a07e0:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x49b7a0}
    FATA[0000] run test [example] err: sql:drop table t1;: run "drop table t1;" at line 2 err Error 1051: Unknown table 'example.t1'

### PR DESCRIPTION
**Pull request by dveeden**
The line numbers in the error messages were not correct.

Closes  #19

This is because `loadQueries()` was sending `[]string` to `ParseQueries()`, which would number them and assign line numbers based on this. However as `loadQueries()` is filtering out lines that start with `#` and combines multi-line statements into a single query this doesn't match with line numbers.

Solution:
Have `loadQueries()` send `[]query` instead of `[]string` to `ParseQueries()`, which has the query string in `query.Query` and the line number in `query.Line`.

Original behavior:
```
$ ./mysql-tester --port=4001 --log-level=debug role
INFO[0000] running tests: [role]                        
WARN[0000] Create new db&{0 {root:@tcp(127.0.0.1:4001)/test?strict=true&time_zone=%27Asia%2FShanghai%27 0x1059168} 0 {0 0} [0xc00016c5a0] map[] 0 1 0xc000096120 false map[0xc00016c5a0:map[0xc00016c5a0:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x49cec0} 
FATA[0000] run test [role] err: sql:CREATE ROLE `ident with space`;: run "CREATE ROLE `ident with space`;" at line 5 err Error 1396: Operation CREATE ROLE failed for 'ident with space'@'%' 
$ sed -n '1,7p' t/role.test | nl
     1	# call mtr.add_suppression(".* Failed to activate default role .*");
     2	--echo # test CREATE ROLE and variations of authorizationID:
     3	--echo # WL988.I-1
     4	--echo # Role is not a reserved keyword
     5	CREATE ROLE `role`;
     6	DROP ROLE `role`;
     7	CREATE ROLE `ident with space`;
```
So it reports `at line 5`, while the statement that has the issue is actually on line 7 of the test file.

With this PR:
```
$ ./mysql-tester --port=4001 --log-level=debug role
INFO[0000] running tests: [role]                        
WARN[0000] Create new db&{0 {root:@tcp(127.0.0.1:4001)/test?strict=true&time_zone=%27Asia%2FShanghai%27 0x1059168} 0 {0 0} [0xc0001b8870] map[] 0 1 0xc0000321e0 false map[0xc0001b8870:map[0xc0001b8870:true]] map[] 0 0 0 0 <nil> 0 0 0 0 0x49cec0} 
FATA[0000] run test [role] err: sql:CREATE ROLE `ident with space`;: run "CREATE ROLE `ident with space`;" at line 7 err Error 1396: Operation CREATE ROLE failed for 'ident with space'@'%' 
[dvaneeden@dve-carbon mysql_test]$ sed -n '7p' t/role.test 
CREATE ROLE `ident with space`;
```

### CODE REVIEW

**Review by morgo — APPROVED** [review]

File: src/main.go [hunk 1]
```diff
@@ -144,16 +144,7 @@ func newTester(name string) *tester {
 
 func setHashJoinConcurrency(db *sql.DB) {
 	if _, err := db.Exec("SET @@tidb_hash_join_concurrency=1"); err != nil {
-		var ignoreErr = false
```
Comment (bb7133) [inline src/main.go:147]: Why ignore error logic is removed?

Comment (morgo) [reply src/main.go:147]: This related to strict mode, and it previously tested if it was a warning here.

Strict mode has been removed from the Go driver: https://github.com/go-sql-driver/mysql/wiki/strict-mode

Comment (dveeden) [reply src/main.go]: I've merged new commits from master in my branch that fix this issue, so it's now gone from the diff.

@bb7133 PTAL

**Review by mjonss — APPROVED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**